01_fetch/src/fetch_noaa_nos.py

```diff
@@ -21,77 +20,53 @@ def fetch_metadata(station_id, metadata_outfile, bucket, write_location, s3_clie
+        config = yaml.safe_load(stream)['fetch_noaa_nos.py']
+        
+    #choose where you want to write location for data outputs
+    write_location = config['write_location']
+    s3_client = utils.prep_write_location(write_location, config['aws_profile'])
+    s3_bucket = config['s3_bucket']
 
-    station_id = '8573927'
+    station_id = config['station_id']
 
-    datum = 'MLLW'
-    ''' options:
-    HAT: Highest Astronomical Tide
-    MHHW: Mean Higher High Water
-    MHW: Mean High Water
-    DTL: Diurnal Tide Level
-    MTL: Mean Tide Level
-    MSL: Mean Sea Level
-    MLW: Mean Low Water 
-    MLLW: Mean Lower Low Water
-    LAT: Lowest Astronomical Tide
-    GT: Great Diurnal Range
-    MN: Mean Diurnal Range
-    DHQ: Mean Diurnal High Water Inequality
-    HWI: Greenwich High Water Interval
-    LWI: Greenwich Low Water Interval
-    Max Tide: Highest Observed Tide
-    Min Tide: Lowest Observed Tide
-    Station Datum: fixed base elevation at a tide station
-    National Tidal Datum Epoch: The specific 19-year period adopted by the National Ocean Service as the official time
-     segment over which tide observations are taken and reduced to obtain mean values (e.g., mean lower low water, etc.) for tidal datums.'''
+    datum = config['datum']
 
-    product = 'water_level'
-    '''options: water_level, air_temperature, water_temperature, wind, air_pressure, air_gap, conductivity, visibility,
-     humidity, salinity, hourly_height, high_low, daily_mean, monthly_mean, one_minute_water_level, predictions, datums,
-     currents, and currents_predictions.'''
+    product = config['product']
 
-    time_zone = 'GMT'
-    '''options:
-    gmt: Greenwich Mean Time
-    lst: Local Standard Time. The time local to the requested station.
-    lst_ldt: Local Standard/Local Daylight Time. The time local to the requested station.'''
+    time_zone = config['time_zone']
 
-    units = 'metric'
-    '''options: english and metric'''
+    units = config['units']
+    # options: english and metric
 
-    file_format = 'csv'
+    file_format = config['file_format']
 
-    start_dt = '20190101'
-    end_dt = '20191231'
+    start_dt = config['start_dt']
+    end_dt = config['end_dt']
 
     path = os.path.dirname('/01_fetch/out/')
     os.path.isdir(path)
-    filename = f"noaa_nos_{product}_{station_id}.csv"
+    filename = "noaa_nos_{products}_{station_id}.csv"
     data_outfile = os.path.join('.', path + '/' + filename)
     fetch_noaa_nos_data(start_dt, end_dt, datum, station_id, time_zone, product, units, file_format, data_outfile, s3_bucket, write_location, s3_client)
 
-    metadata_filename = f"noaa_nos_metadata_{product}_{station_id}.csv"
+    path = os.path.dirname('/01_fetch/out/')
+    metadata_filename = f"noaa_nos_metadata_{products}_{station_id}.csv"
     metadata_outfile = os.path.join('.', path + '/' + metadata_filename)
+    metadata_outfile = os.path.join('.' + '/' + metadata_filename)
```

Comment:
I think this and the previous line are overwriting each other - you just need one. I would prefer the first line that sets the path relative to the home directory (drb-estuary-salinity), and we just make sure we are working in that directory when we execute the code. THat's how our other codes are set up so far. Let me know if you can't get this to run with the previous line  of code instead of this one, and we can try to debug!
```suggestion
```

Reply:
Makes sense. Yes. I was having trouble changing the current directory with os in Jupyter (erggg), so I commented out the previous line and set it to the current directory. I forgot to remove it. Thanks for catching that.